dev/breeze/src/airflow_breeze/utils/docker_command_utils.py

```diff
@@ -804,23 +805,34 @@ def autodetect_docker_context():
 
     :return: name of the docker context to use
     """
-    output = run_command(["docker", "context", "ls", "-q"], capture_output=True, check=False, text=True)
-    if output.returncode != 0:
+    result = run_command(
+        ["docker", "context", "ls", "--format=json"],
```

Comment:
I tried this before, and it's better to stick to "ls -q".

Unfortunately there are still relatively fresh versions of docker out there that do not support `--format` for this command. We had some cases like that where people's docker (even newly installed on relatively fresh Linux machine) did not contain the `--format` switch.

Reply:
Maybe actually we could leave it as is but raise the min-docker version requirement though (but we need to figure out in which version of docker it has been added. I think evan if it is relatively recent addition, we can actually raise the bar a little as we have not updated the min docker version for quite a while. The worst thing, the users will just have to update their docker following instructions on docker website (and I think we already direct the users there when min-version is not reached).

Reply:
I did some experimenting on my Linux machine (reinstalled some older versions and bisected when the change happened as it was not in the release notes):

Those are versions available for my Mint at least:

```
5:24.0.5-1~ubuntu.22.04~jammy
5:24.0.4-1~ubuntu.22.04~jammy
5:24.0.3-1~ubuntu.22.04~jammy
5:24.0.2-1~ubuntu.22.04~jammy
5:24.0.1-1~ubuntu.22.04~jammy
5:24.0.0-1~ubuntu.22.04~jammy
5:23.0.6-1~ubuntu.22.04~jammy
5:23.0.5-1~ubuntu.22.04~jammy
5:23.0.4-1~ubuntu.22.04~jammy
5:23.0.3-1~ubuntu.22.04~jammy
5:23.0.2-1~ubuntu.22.04~jammy
5:23.0.1-1~ubuntu.22.04~jammy
5:23.0.0-1~ubuntu.22.04~jammy
5:20.10.24~3-0~ubuntu-jammy
5:20.10.23~3-0~ubuntu-jammy
5:20.10.22~3-0~ubuntu-jammy
5:20.10.21~3-0~ubuntu-jammy
5:20.10.20~3-0~ubuntu-jammy
5:20.10.19~3-0~ubuntu-jammy
5:20.10.18~3-0~ubuntu-jammy
5:20.10.17~3-0~ubuntu-jammy
5:20.10.16~3-0~ubuntu-jammy
5:20.10.15~3-0~ubuntu-jammy
5:20.10.14~3-0~ubuntu-jammy
5:20.10.13~3-0~ubuntu-jammy
```

For 5:20.10.24~3-0~ubuntu-jammy (our current min-version is 20.10.0):
```
[jarek@jarek-HP-Z2-Tower-G9-Workstation-Desktop-PC:~/code/airflow] main+ 29s ± docker context ls --help

Usage:  docker context ls [OPTIONS]

List contexts

Aliases:
  ls, list

Options:
      --format string   Pretty-print contexts using a Go template
  -q, --quiet           Only show context names
```

For 5:23.0.0-1~ubuntu.22.04~jammy:

```
Usage:  docker context ls [OPTIONS]

List contexts

Aliases:
  docker context ls, docker context list

Options:
      --format string   Format output using a custom template:
                        'table':            Print output in table format with column headers (default)
                        'table TEMPLATE':   Print output in table format using the given Go template
                        'json':             Print in JSON format
                        'TEMPLATE':         Print output using the given Go template.
                        Refer to https://docs.docker.com/go/formatting/ for more information about formatting output with templates
  -q, --quiet           Only show context names
```

I'd be for setting the min version to `23.0.6` ? WDYT @uranusjr ?

Reply:
23.0.6[🔗](https://docs.docker.com/engine/release-notes/23.0/#2306) `2023-05-08`

Released 3 months ago.

I think for a development environment it's sortof enough. We can always experiment and go back if people will complain. 

We could also raise the docker-compose min version - but that should be done separately, as we would have to force docker-compose v2. Which I think it's about the time to do anyway.

Reply:
Hmm since 20.10.24 already supports `--format` with a Go template, _and_ that syntax is still supported in 23.0, I think it might be better to just use that syntax instead. I’ll work on that.

Reply:
That is also an option indeed.

Reply:
But I think it's about right to bump min-version of docker. This is a small annoyance for contributors to upgrade, but it's good for contributors to update and refresh docker client, there might be other things in the future that we won't even be aware that we are lagging behind. If you want I can make a separate PR where I bump both docker-compose  (we still support v1 which is > 2 years old) and docker min versions. This is already handled well by messaging in Breeze.